src/compute/src/sink/persist_sink.rs

```diff
@@ -0,0 +1,198 @@
+        // handling multiple in-flight write futures, similar to how the persist operators in
+        // `operators/stream.rs` do it. That would allow us to have multiple write requests in
+        // flight concurrently.
+        persist_op.build_async(
+            scope,
+            move |_capabilities, frontiers, scheduler| async move {
+                let mut buffer = Vec::new();
+                let mut stash: HashMap<Timestamp, Vec<((Row, Row), Timestamp, Diff)>> =
+                    HashMap::new();
+
+                // Keep an empty Row that we can re-use when we are replacing an Option with an empty
+                // Row. Our output type is `(Row, Row)` instead of `(Option<Row>, Option<Row>)`.
+                let empty_row = Row::default();
+
+                while scheduler.notified().await {
+                    let frontier = frontiers.borrow()[0].clone();
+
+                    if !active_write_worker {
+                        return;
+                    }
+
+                    let mut write = write.borrow_mut();
+                    let write = write.deref_mut();
+                    let write = match write {
+                        Some(write) => write,
+                        None => {
+                            // We have been cancelled!
+                            return;
+                        }
+                    };
+
+                    input.for_each(|_cap, data| {
+                        data.swap(&mut buffer);
+
+                        let mapped_updates = buffer.drain(..).map(|((key, value), ts, diff)| {
+                            let key = key.unwrap_or_else(|| empty_row.clone());
+                            let value = value.unwrap_or_else(|| empty_row.clone());
+                            ((key, value), ts, diff)
+                        });
+
+                        for update in mapped_updates {
+                            stash.entry(update.1).or_default().push(update);
+                        }
+                    });
+
+                    let frontier1 = &frontier;
+                    let updates = stash
+                        .iter()
+                        .filter(move |(ts, _updates)| !frontier1.less_equal(ts))
+                        .flat_map(|(_ts, updates)| updates.iter())
+                        .map(|((key, value), ts, diff)| ((key, value), ts, diff));
+
+                    // We always append, even in case we don't have any updates, because appending
+                    // also advances the frontier.
+                    // TODO(aljoscha): Figure out how errors from this should be reported.
+                    write
+                        .append(timeout, updates, frontier.clone())
+                        .await
+                        .expect("cannot append updates")
+                        .expect("cannot append updates");
+
+                    let frontier1 = &frontier;
```

Comment:
Out of curiosity, what is the reason for creating these `frontier1` variables?

Reply:
These are leftovers from when I had to `clone()` them in an earlier version of the code, I'll get rid of them